### DISCUSSION TITLE
Pretrained AI

### DISCUSSION THREAD

**Discussion by justin-kumbroch** [trackmania-rl/tmrl#55] [Q&A] [answered]
Would it be possible to get the pretrained AI mentioned in the TLDR? I will try to carry on the work of @donadigo and experiment with a map-building AI. I thought of using the driving Agent to test driveability.

Best regards

**Comment by yannbouteiller** [comment, marked as answer]
Hi, sure, the agent is provided as part of the library, all you need to do is follow the [tutorial](https://github.com/trackmania-rl/tmrl/blob/master/readme/get_started.md). 

The pre-trained agent will only drive on the tmrl-test track though. If you want to use other tracks, you will need to train new agents.

**Reply by Drefrank** [reply]
thats cool but difficult to

**Reply by justin-kumbroch** [reply]
Ok, thank you so much!

**Reply by justin-kumbroch** [reply]
Do you know of a driving Agent that is capable of navigating through novel tracks (i.e. driving without being trained on the track)?

**Reply by yannbouteiller** [reply]
Not really, this would be very hard to train.

**Reply by eliegoudout** [reply]
I will add that you cannot expect, in general, zero shot very good driving on novel tracks, as it might not be possible to anticipate depending on whether or not parts of the track are hidden by wall behind turns, for example.

**Reply by justin-kumbroch** [reply]
Okay, thank you for your input!
